src/support/modify.c

```diff
@@ -470,8 +470,7 @@ __wt_modify_reconstruct_from_upd_list(
     __wt_update_vector_init(session, &modifies);
     /* Find a complete update. */
     for (upd = modify; upd != NULL; upd = upd->next) {
-        if (upd->txnid == WT_TXN_ABORTED)
-            continue;
+        WT_ASSERT(session, upd->type != WT_UPDATE_TOMBSTONE);
```

Comment:
Shall we also assert the update is not aborted?

Reply:
This turned out to be a rabbit hole - there is other code that can append updates besides what I fixed. So I limited the scope of this ticket to fixing {{__btcur_update}}, which is the function that pushed the bad update in the original test failure. I made WT-12008 for fixing the other code.